msfinder wrapper

zargham-ahmad opened the pull request:
This PR contains the galaxy wrapper for the recetox_msfinder.

Todo:
- [x] pushed docker image to docker hub
- [x] update the docker image in requirement

@hechth can you also check if the description in the .shed.yml and in the tool is okay

closes #292

hechth commented:
@zargham-ahmad Can you add the `.shed.yml` file and rename the tool to only `recetox-msfinder`? The october prefix on the xMSannotator tool was since it was a prototype that was due in october - normally there is no need to add some prefix related to the date of publication etc.

zargham-ahmad commented:
> @zargham-ahmad Can you add the `.shed.yml` file and rename the tool to only `recetox-msfinder`? The october prefix on the xMSannotator tool was since it was a prototype that was due in october - normally there is no need to add some prefix related to the date of publication etc.

I have added the .shed.yml file, if you could verify it that will be great. I have updated the tool name but the tool file name should be based on the docker image and needs to be updated based on the docker image tag (currently based on the temporary image), also the docker image name need to be updated as well (currently based on the temporary image)

zargham-ahmad commented on tools/recetox_msfinder/.shed.yml:
```diff
@@ -0,0 +1,11 @@
+    description: "Peak Annotation"
+    include:
+      - november_recetox_msfinder.xml
+      - macros.xml
```
resolved in c28b5c1

bgruening replied:
The shed.yml is only a metadata file for the ToolShed. Its not really taken into account during normal testing.

bgruening replied:
If its not really needed, I would not use include, or exclude to avoid failures in the TS repo.

hechth replied:
The tools are also being pushed to the main toolshed by the CI, so I think all tools should have a .shed.yaml file to add the required metadata - I think otherwise it is also not possible to upload them there?

bgruening commented on tools/recetox_msfinder/recetox_msfinder.xml:
```diff
@@ -0,0 +1,70 @@
+    <command>
+        <![CDATA[
+        sh ${write_param} &&
+        mono /MsFinder/MsFinder/bin/Debug/MsfinderConsoleApp.exe annotate -i "$input_data" -m "MSFINDER.INI" -o "$annotated_data"        
```
please use single-quotes around all path-like or text-like parameters

zargham-ahmad replied:
resolved in [feb89bf](https://github.com/RECETOX/galaxytools/pull/313/commits/feb89bf7c3effb19096caadbb7de383cc3254627)

bgruening commented on tools/recetox_msfinder/recetox_msfinder.xml:
```diff
@@ -0,0 +1,70 @@
+            <output name="annotated_data" file="out.msp" lines_diff="16" />
+        </test>
+    </tests>
+
```
help is missing

hechth replied:
A help text was added now.

maximskorik commented on tools/recetox_msfinder/macros.xml:
```diff
@@ -0,0 +1,60 @@
+        </param>
+    </xml>
+    <xml name="parameter">
+        <param type="float" value="0.01" name="Ms2Tolerance" label="Ms2Tolerance" help="The mass tolerance for matching experimental- and reference fragments." />
```
I'd suggest something like "Fragments m/z tolerance" for a label.

zargham-ahmad replied:
@hechth can you share your thoughts on this

hechth replied:
I think this should be mass tolerance used for fragment peak matching and annotation.

maximskorik commented on tools/recetox_msfinder/macros.xml:
```diff
@@ -0,0 +1,60 @@
+        </param>
+    </xml>
+    <xml name="parameter">
+        <param type="float" value="0.01" name="Ms2Tolerance" label="Ms2Tolerance" help="The mass tolerance for matching experimental- and reference fragments." />
```
I don't think that hyphen is necessary.
```suggestion
        <param type="float" value="0.01" name="Ms2Tolerance" label="Ms2Tolerance" help="The mass tolerance for matching experimental and reference fragments." />
```

zargham-ahmad replied:
resolved in [feb89bf](https://github.com/RECETOX/galaxytools/pull/313/commits/feb89bf7c3effb19096caadbb7de383cc3254627)

maximskorik commented on tools/recetox_msfinder/macros.xml:
```diff
@@ -0,0 +1,60 @@
+            help="Check TMS-MeOX derivative to use EIMS database; otherwise, MSMS database is used." />
+        <param type="integer" value="1" name="RelativeAbundanceCutOff" label="RelativeAbundanceCutOff"
+            help="The product ions more than this parameter on the basis of base peak ion are utilized for the product ion matching." />
+        <param type="float" value="0.001" name="Ms1Tolerance" label="Ms1Tolerance" help="The mass tolerance to generate formula candidates." />
```
"Precursor m/z tolerance" could be used as a label.

maximskorik commented on tools/recetox_msfinder/macros.xml:
```diff
@@ -0,0 +1,60 @@
+        <param type="float" value="0.01" name="Ms2Tolerance" label="Ms2Tolerance" help="The mass tolerance for matching experimental- and reference fragments." />
+        <param name="IsTmsMeoxDerivative" type="boolean" checked="False" truevalue="True" falsevalue="False" label="IsTmsMeoxDerivative"
+            help="Check TMS-MeOX derivative to use EIMS database; otherwise, MSMS database is used." />
+        <param type="integer" value="1" name="RelativeAbundanceCutOff" label="RelativeAbundanceCutOff"
+            help="The product ions more than this parameter on the basis of base peak ion are utilized for the product ion matching." />
```
I'm a bit confused by this help section: do fragments with abundance above or below this value get filtered out (not used)? 

As it seems from the description, those that are below this value are not used. In this case, however, the default value of `1` means no ions will be used for matching because no ion can have relative intensity (to a base peak) higher than `1`.

hechth replied:
```suggestion
        <param type="integer" value="1" name="RelativeAbundanceCutOff" label="RelativeAbundanceCutOff"
            help="Remove peaks with less then specified % relative intensity from the spectra." />
```

maximskorik reviewed (changes requested):
@zargham-ahmad, looks good, and you did a great job figuring out the input parameters and their interpretation! Please polish and reformat the labels of the parameters a bit to make them more user-readable before merging.

hechth commented on tools/recetox_msfinder/macros.xml:
```diff
@@ -0,0 +1,60 @@
+    </xml>
+    <xml name="input">
+        <param name="input_data" type="data" format="msp,mat">
+            <label>Input file to process</label>
```
```suggestion
            <label>Input mass spectral library file to which to add peak annotations.</label>
```

hechth commented on tools/recetox_msfinder/macros.xml:
```diff
@@ -0,0 +1,60 @@
+        </creator>
+    </xml>
+    <xml name="input">
+        <param name="input_data" type="data" format="msp,mat">
```
```suggestion
        <param name="input_data" type="data" format="msp">
```

Do we really support `mat` at the moment? If so, there should be a test for it. I'd vote for rather not using this format.

hechth replied:
Also, it should be stated that Formula, InChI and SMILES codes are required to be present in the library file.

zargham-ahmad replied:
resolved in 50fcbda

hechth commented on tools/recetox_msfinder/macros.xml:
```diff
@@ -0,0 +1,60 @@
+            help="The product ions more than this parameter on the basis of base peak ion are utilized for the product ion matching." />
+        <param type="float" value="0.001" name="Ms1Tolerance" label="Ms1Tolerance" help="The mass tolerance to generate formula candidates." />
+        <param type="select" name="MassToleranceType" label="MassToleranceType">
+            <option value="Da" selected="True">Da</option>
+            <option value="Ppm">Ppm</option>
```
They could use a help text, with `ppm` standing for `parts per million` and `Da` standing for Daltons (amu).

zargham-ahmad replied:
resolved in 50fcbda

hechth commented on tools/recetox_msfinder/macros.xml:
```diff
@@ -0,0 +1,60 @@
+            <option value="Da" selected="True">Da</option>
+            <option value="Ppm">Ppm</option>
+        </param>
+        <param type="integer" value="-1" name="StructurePredictionTimeOut" label="StructurePredictionTimeOut" />
```
This needs a help text - what does a value of `-1` mean? Maybe this could also be made optional.

zargham-ahmad replied:
resolved in 50fcbda

hechth commented on tools/recetox_msfinder/recetox_msfinder.xml:
```diff
@@ -0,0 +1,70 @@
+<tool id="recetox_msfinder" name="RECETOX MsFinder" version="@TOOL_VERSION@+galaxy0">
+    <description>Peak Annotation</description>
```
```suggestion
    <description>Annotation of fragment peaks in mass spectral libraries.</description>
```

hechth commented on tools/recetox_msfinder/.shed.yml:
```diff
@@ -0,0 +1,11 @@
+repositories:
+  recetox_msfinder:
+    description: "MS-FINDER is program for compound annotation for EI-MS (GC/MS) and MS/MS data"
+    include:
+      - recetox_msfinder.xml
+      - macros.xml
```
Maybe also include the test data and the shed.yaml file?

zargham-ahmad replied:
resolved in b3c4c32

hechth reviewed (commented):
Some things have to be improved regarding documentation and help text. Also a general description of the tool (as Bjoern pointed out) would be good.

hechth commented on tools/recetox_msfinder/macros.xml:
```diff
@@ -54,6 +54,24 @@
+
+        The input is a mass spectral library file to which to add peak annotations.
+
+        Note: Formula, InChI and SMILES codes are required to be present in the input mass spectral library file.
```
```suggestion
        Note: Formula and InChI (or SMILES) codes are required to be present in the input mass spectral library file.
```
Sorry, thought that both is required but it also works with only one of InChI or SMILES

hechth commented on tools/recetox_msfinder/macros.xml:
```diff
@@ -0,0 +1,78 @@
+        Prof. Oliver Fiehn team (UC Davis) supported by the JST/NSF SICORP Metabolomics for the low carbon society 
+        project.
+
+        The input is a mass spectral library file to which to add peak annotations.
+
+        Note: Formula and InChI (or SMILES) codes are required to be present in the input mass spectral library file.
```
```suggestion
        This galaxy wrapper only peforms MS2 peak annotation given a correctly formatted mass spectral library file.
        FORMULA, PRECURSORMZ and SMILES (or INCHI) metadata entries are required to be present in the input mass spectral library file.
        Each spectrum has to be separated by a single blank line.
```

maximskorik commented on tools/recetox_msfinder/macros.xml:
```diff
@@ -0,0 +1,78 @@
+        <param name="IsTmsMeoxDerivative" type="boolean" checked="False" truevalue="True" falsevalue="False" label="IsTmsMeoxDerivative"
+            help="Check TMS-MeOX derivative to use EIMS database; otherwise, MSMS database is used." />
+        <param type="integer" value="1" name="RelativeAbundanceCutOff" label="RelativeAbundanceCutOff"
+            help="Remove peaks with less then specified % relative intensity from the spectra." />
```
A little typo:
```suggestion
            help="Remove peaks with less than specified % relative intensity from the spectra." />
```

hechth commented on tools/recetox_msfinder/macros.xml:
```diff
@@ -0,0 +1,78 @@
+            help="Check TMS-MeOX derivative to use EIMS database; otherwise, MSMS database is used." />
+        <param type="integer" value="1" name="RelativeAbundanceCutOff" label="RelativeAbundanceCutOff"
+            help="Remove peaks with less than specified % relative intensity from the spectra." />
+        <param type="float" value="0.001" name="Ms1Tolerance" label="Ms1Tolerance" help="The mass tolerance to generate formula candidates." />
```
```suggestion
        <param type="float" value="0.001" name="Ms1Tolerance" label="Ms1Tolerance" help="The precursor m/z tolerance to generate formula candidates." />
```

xtrojak commented on tools/recetox_msfinder/macros.xml:
```diff
@@ -0,0 +1,78 @@
+<macros>
+    <token name="@TOOL_VERSION@">0.1.0</token>
```
I suppose the wrapper version should reflect the tool version, similarly to docker container.

```suggestion
    <token name="@TOOL_VERSION@">3.5.2</token>
```

xtrojak commented on tools/recetox_msfinder/recetox_msfinder.xml:
```diff
@@ -0,0 +1,76 @@
+    <tests>
+        <test>
+            <param name="input_data" value="test.msp" ftype="msp" />
+            <output name="annotated_data" file="out.msp" lines_diff="16" />
```
What is the purpose of this test with such tolerance? Are we planning to improve that? Shouldn't that be reported somewhere?

xtrojak reviewed (changes requested):
Still some minor issues.
